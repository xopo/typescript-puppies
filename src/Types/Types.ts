export interface APuppy {
    id: number;
    name: string;
    type: string;
    adopted: boolean;
}

export type Puppies = Array<APuppy>

export type NewPuppy = APuppy & { id?: number};

export const ADD_PUPPY = 'ADD_PUPPY';

export const ADD_REQUEST = 'ADD_REQUEST';

export const RECEIVED_REQUEST = 'RECEIVED_REQUEST';

export const GET_PUPPIES = 'GET_PUPPIES';
