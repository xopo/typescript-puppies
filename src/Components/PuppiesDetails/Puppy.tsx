import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import { APuppy } from '../../Types/Types';

const StyledPuppy = styled.div`
    display: flex;
    flex-direction: column;
    align-items:flex-start;
    border: 2px dashed black;
    margin: 5px 5px auto;
    padding: 5px;
    width: 30%;
    div {
        display: block
    }
`;


const  Puppy:React.FC<APuppy> = props => {
    const { id, name, type, adopted } = props;
    return(
        <StyledPuppy>
            <div>Name: <Link title={`navigate to ${name}`} to={`/${id}`}>{name}</Link></div>
            <div>Type: {type}</div>
            <div>Adopted: <b>{adopted}</b></div>
        </StyledPuppy>
   )
};

export default Puppy;