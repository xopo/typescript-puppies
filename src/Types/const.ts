export const ADD_PUPPY = 'ADD_PUPPY';

export const GET_PUPPIES = 'GET_PUPPIES';

export const DELETE_PUPPY = 'DELETE_PUPPY';

export const ADOPT_PUPPY = 'ADOPT_PUPPY';

export const PUPPIES_GET_REQUEST = 'PUPPIES_GET_REQUEST';

export const PUPPIES_GET_SUCCESS = 'PUPPIES_GET_SUCCESS';