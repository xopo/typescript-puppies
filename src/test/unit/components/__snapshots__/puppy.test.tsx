import React from 'react';
import { shallow } from 'enzyme';
import Puppy from '../../../../Components/PuppiesDetails/Puppy';
import { APuppy } from '../../../../Types/Types';

const adopt = jest.fn();
const del = jest.fn();
const props: APuppy = {
  id: 3,
  adopted: false,
  name: 'Bob',
  type: 'canis',
  adopt,
  delete: del
}

describe('Puppy Component', () => {
  it('should render witout errors', () => {
    const shallowElement = shallow<APuppy>(<Puppy {...props}/>);
    
    expect(shallowElement).toMatchSnapshot();
  });

  it('should trigger adopt function when adopt button is clicked and send the id as param', () => { 
    const shallowElement = shallow<APuppy>(<Puppy {...props}/>);
    const adoptButton = shallowElement.find('.adopt');
    adoptButton.simulate('click');
    
    expect(adoptButton.length).toBe(1);
    expect(adopt).toHaveBeenCalled();
    expect(adopt).toHaveBeenCalledWith(props.id);
  });
}) 