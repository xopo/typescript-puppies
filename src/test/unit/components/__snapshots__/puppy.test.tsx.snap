// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puppy Component should render witout errors 1`] = `
<styled.div>
  <div>
    Name: 
    <Link
      title="navigate to Bob"
      to="/3"
    >
      Bob
    </Link>
  </div>
  <div>
    Type: 
    canis
  </div>
  <div>
    Adopted: 
    <b>
      false
    </b>
  </div>
  <form>
    <button
      className="adopt"
      onClick={[Function]}
      type="button"
    >
      Adopt
    </button>
    <button
      className="delete"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </form>
</styled.div>
`;
